### PR TITLE
docs: use quantinuum-sphinx theme

### PR DESCRIPTION
**Pull request by daniel-mills-cqc**
I've created this new branch with your changes on top of main. Let me know if this is actually helpful.

### CODE REVIEW

File: .github/workflows/deploy_docs.yml
Comment (daniel-mills-cqc) [inline .github/workflows/deploy_docs.yml:1]: I think that the path to the artifact to be uploaded will need to be changed to he build directory.

**Comment by CalMacCQ** [comment]
thanks for sorting this out, I should be able to finish this off today

File: pyproject.toml [hunk 1]
```diff
@@ -22,10 +22,10 @@ pytest = {version="^8.1.1", optional=true}
 mypy = {version="^1.9.0", optional=true}
 sphinx = {version="^7.2.6", optional=true}
 pytest-cov = {version="^4.1.0", optional=true}
-sphinx-book-theme = {version="^1.1.2", optional=true}
 qiskit-ibm-provider = {version="^0.10.0", optional=true}
 pytket-quantinuum = {version="0.40.0", optional=true}
 ruff = {version="^0.7.4", optional=true}
+furo = {version="^2024.8.6", optional=true}
```
Comment (CalMacCQ) [inline pyproject.toml:28]: Would you prefer this under "extras"?

File: .github/workflows/deploy_docs.yml
Comment (CalMacCQ) [reply .github/workflows/deploy_docs.yml:1]: hopefully fixed in [44e823d](https://github.com/CQCL/Qermit/pull/120/commits/44e823d67fcb41aa8d2b5b70dce06875aed65d8b)

File: docs/_static/nav-config.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const navConfig = {
+    "navTextLinks": [
+        {
+            "title": "API docs",
+            "href": "index.html",
+        },
+        {
+            "title": "Manual",
+            "href": "manual/manual_index.html",
```
Comment (CalMacCQ) [inline docs/_static/nav-config.js:9]: By the way these links are configurable (at least with this version of quantinuum-sphinx) 

feel free to suggest other places for the navbar to go

**Review by daniel-mills-cqc — COMMENTED** [review]
Looking great!
Would you mind checking the warning from the build. I see a warning
```
WARNING: html_static_path entry 'quantinuum-sphinx/_static' does not exist
WARNING: favicon file 'quantinuum-sphinx/_static/assets/quantinuum_favicon.svg' does not exist
```
Could those be fixed?

**Comment by CalMacCQ** [comment]
> Looking great! Would you mind checking the warning from the build. I see a warning
> 
> ```
> WARNING: html_static_path entry 'quantinuum-sphinx/_static' does not exist
> WARNING: favicon file 'quantinuum-sphinx/_static/assets/quantinuum_favicon.svg' does not exist
> ```
> 
> Could those be fixed?

Ah thanks, I can fix that

**Comment by CalMacCQ** [comment]
> Looking great! Would you mind checking the warning from the build. I see a warning
> 
> ```
> WARNING: html_static_path entry 'quantinuum-sphinx/_static' does not exist
> WARNING: favicon file 'quantinuum-sphinx/_static/assets/quantinuum_favicon.svg' does not exist
> ```
> 
> Could those be fixed?

hmm I'm not getting those.. Perhaps a submodule issue?

Try doing

```shell
git submodule update --init --recursive
```

Also make sure you're on commit `a8ac27e089824099708da2cea9cef262e4df3ad2 quantinuum-sphinx (heads/dist)`

run this to check

```shell
git submodule status
```

**Comment by CalMacCQ** [comment]
I've also added a README in the `docs` folder. Let me know if you're happy with this. 

Would you recommend using poetry or pip in local build and github worklfows?

File: pyproject.toml (hunk 1 above)
Comment (CalMacCQ) [reply pyproject.toml]: Done, let me know if I got it right

**Comment by CalMacCQ** [comment]
> Could those be fixed?

Warnings now fixed, hadn't checked out the submodule in C.I. properly.

These warnings were actually quite serious. I've added a `-W` flag to fail the sphinx C.I. build on warnings to catch this sort of thing in future.

File: pyproject.toml [hunk 3]
```diff
@@ -18,18 +18,25 @@ python = ">=3.10, <3.13"
...
+qiskit-ibm-provider = { version = "^0.10.0", optional = true }
+pytket-quantinuum = { version = "0.40.0", optional = true }
+ruff = { version = "^0.7.4", optional = true }
+furo = { version = "^2024.8.6", optional = true }
 
 [tool.poetry.extras]
-tests = ["pytest", "mypy", "ruff", "pytest-cov", "qiskit-ibm-provider", "pytket-quantinuum"]
-docs = ["sphinx", "sphinx-book-theme", "pytket-quantinuum"]
+tests = [
+    "pytest",
```
Comment (CalMacCQ) [inline pyproject.toml:32]: think my linter made these changes?

**Comment by daniel-mills-cqc** [comment]
> I've also added a README in the `docs` folder. Let me know if you're happy with this.
> 
> Would you recommend using poetry or pip in local build and github workflows?

Thanks, looking good!

I don't really recommend using either in particular, but I'm happy to stick with pip in this readme.

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -33,20 +33,19 @@ jobs:
...
-          cd docs_src
-          ./build_docs.sh
-          cd ../manual
-          ./build_manual.sh
+          cd docs
+          ./build-docs.sh
       - name: Save documentation
         uses: actions/upload-artifact@v4
         with:
-          name: docs_html
```
[the remark this answers is not in the document]
Comment (daniel-mills-cqc) [reply .github/workflows/build_and_test.yml:49]: I don't think it matters hugely, but I think only the build itself really needs uploading here. Would you mind making that change?

**Comment by daniel-mills-cqc** [comment]
> > Could those be fixed?
> 
> Warnings now fixed, hadn't checked out the submodule in C.I. properly.
> 
> These warnings were actually quite serious. I've added a `-W` flag to fail the sphinx C.I. build on warnings to catch this sort of thing in future.

Locally I am still getting
```
html_static_path entry 'quantinuum-sphinx/_static' does not exist
```
Does that surprise you?
I think I need to clone the quantinuum-sphinx subodule. If I clone -> https://github.com/CQCL/quantinuum-sphinx/tree/a8ac27e089824099708da2cea9cef262e4df3ad2 into the `quantinuum-sphinx` directory will that do the job?

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+git clone git@github.com:CQCL/Qermit.git --recurse-submodules
+```
+
+2. Install Qermit and docs dependencies
+
+```shell
+python -m venv ".venv"
+source .venv/bin/activate
+pip install -e .[docs] 
+```
```
Comment (daniel-mills-cqc) [inline docs/README.md:15]: I think we could remove the reference to venv and just allow the user to use whichever means they prefer.
You could also mention at this point that poetry is supported.

**Review by daniel-mills-cqc — COMMENTED** [review]
Just this last small comment, then it's good to go!

File: pyproject.toml (hunk 1 above)
Comment (daniel-mills-cqc) [reply pyproject.toml]: Looks good to me! Thanks!

File: .github/workflows/build_and_test.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (CalMacCQ) [reply .github/workflows/build_and_test.yml:49]: Fixed now

**Pull request merged**